Replicator/IncomingRev.cc

```diff
@@ -277,6 +352,46 @@ namespace litecore { namespace repl {
             return;
         }
 
+        std::vector<PendingBlob>::const_iterator danglingBlobBegin = _pendingBlobs.end();
+        if (attachmentsFromSG.has_value()) {
+            vector<PendingBlob> dangled;
+            for (auto iter = _pendingBlobs.begin(); iter != _pendingBlobs.end();) {
+                auto digest = iter->key.digestString();
+                if (attachmentsFromSG->find(digest) == attachmentsFromSG->end()) {
+                    dangled.push_back(*iter);
+                    iter = _pendingBlobs.erase(iter);
+                } else {
+                    ++iter;
+                }
+            }
+            for (const auto& b: dangled) {
+                _pendingBlobs.push_back(b);
+            }
+            _blob = _pendingBlobs.begin();
+            danglingBlobBegin = _blob + (_pendingBlobs.size() - dangled.size());
+        }
```

Comment:
We could simplify this logic by putting the dangled attachments in a separated vector.

Reply:
done